overlord/devicestate/firstboot_test.go

```diff
@@ -1296,12 +1296,16 @@ snaps:
 	chg1 := st.NewChange("become-operational", "init device")
 	chg1.SetStatus(state.DoingStatus)
 
-	st.Unlock()
-	err = s.overlord.Settle(settleTimeout)
-	st.Lock()
-	c.Assert(err, IsNil)
-
-	// at this point snapd is "restarting", pretend the restart has
+	// run change until it wants to restart
+	for i := 0; i < 25; i++ {
+		st.Unlock()
+		s.overlord.Settle(100 * time.Millisecond)
```

Comment:
maybe I'm confused but using Settle this way is a bit strange, it will timeout, no?

Reply:
Thanks, I looked into this and I think the issue is that we return a retry error in auto-connect now. This means that settle never converges because there is always a task scheduled to run (the retry). At least thats my understanding of how settle works.